File: src/types/email.ts
export interface EmailSummary {
  id: string;
  account_id: string;
  uid: number;
  subject: string;
  sender_name: string | null;
  sender_email: string | null;
  received_at: string | null;
  codes: string[];
  is_read: boolean;
}

export interface EmailDetail extends EmailSummary {
  body_text: string | null;
  body_html: string | null;
}

export interface AccountUnreadState {
  account_id: string;
  unread_count: number;
}

export interface NewMailEvent {
  accountId: string;
  emailId: string;
  subject: string;
  sender: string;
  code?: string | null;
  unreadCount: number;
}

File: src-tauri/src/models/email.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EmailSummary {
    pub id: String,
    pub account_id: String,
    pub uid: i64,
    pub subject: String,
    pub sender_name: Option<String>,
    pub sender_email: Option<String>,
    pub received_at: Option<String>,
    pub codes: Vec<String>,
    pub is_read: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EmailDetail {
    pub id: String,
    pub account_id: String,
    pub uid: i64,
    pub subject: String,
    pub sender_name: Option<String>,
    pub sender_email: Option<String>,
    pub received_at: Option<String>,
    pub body_text: Option<String>,
    pub body_html: Option<String>,
    pub codes: Vec<String>,
    pub is_read: bool,
}

#[derive(Debug, Clone)]
pub struct StoredEmail {
    pub id: String,
    pub account_id: String,
    pub uid: i64,
    pub message_id: Option<String>,
    pub subject: String,
    pub sender_name: Option<String>,
    pub sender_email: Option<String>,
    pub received_at: Option<String>,
    pub body_text: Option<String>,
    pub body_html: Option<String>,
    pub codes: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
pub struct AccountUnreadState {
    pub account_id: String,
    pub unread_count: i64,
}
